### PR TITLE
spawn an arangosh that checks cluster nodes are responsive

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/397/

File: js/client/modules/@arangodb/test-utils.js [hunk 1]
```diff
@@ -407,6 +407,8 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
   }
   results.shutdown = results.shutdown && pu.shutdownInstance(instanceInfo, clonedOpts, forceTerminate);
 
+  loadClusterTestStabilityInfo(results, instanceInfo);
```
Comment (KVS85) [inline js/client/modules/@arangodb/test-utils.js]: Where is it being written and when?

File: js/client/modules/@arangodb/clusterstats.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+const download = internal.download;
+const time = internal.time;
+const sleep = internal.sleep;
+
+const pu = require('@arangodb/process-utils');
+
+const instanceInfo = JSON.parse(internal.env.INSTANCEINFO);
+const options = JSON.parse(internal.env.OPTIONS);
+const outfn = fs.join(instanceInfo.rootDir, 'stats.jsonl');
```
Comment (KVS85) [inline js/client/modules/@arangodb/clusterstats.js]: Why it doesn't use `clusterHealthMonitorFile`?

File: js/client/modules/@arangodb/clusterstats.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  instanceInfo.arangods.forEach(arangod => {
+    let serverId = arangod.role + '_' + arangod.port;
+    let beforeCall = time();
+
+    if (arangod.role === "agent") {
+      let reply = download(arangod.url + '/_api/version', '', opts);
+      if (reply.hasOwnProperty('error') || reply.code !== 200) {
+        print(reply);
+        state.state = false;
```
Comment (KVS85) [inline js/client/modules/@arangodb/clusterstats.js]: Why similar naming is used for nested property?

File: js/client/modules/@arangodb/clusterstats.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+const instanceInfo = JSON.parse(internal.env.INSTANCEINFO);
+const options = JSON.parse(internal.env.OPTIONS);
+const outfn = fs.join(instanceInfo.rootDir, 'stats.jsonl');
+const opts = Object.assign(pu.makeAuthorizationHeaders(options),
+                           { method: 'GET' });
+
+while(true) {
+  const before = time();
+  let state = {
+    state: true,
```
Comment (KVS85) [inline js/client/modules/@arangodb/clusterstats.js]: Why similar naming is used for nested property?

File: js/client/modules/@arangodb/clusterstats.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          state: true,
+          start: beforeCall,
+          delta: time() - beforeCall,
+          uptime: statisticsReply.server.uptime
+        };
+      }
+    }
+  });
+  state['delta'] = time() - before;
+  if (state.delta > 1000) {
```
Comment (KVS85) [inline js/client/modules/@arangodb/clusterstats.js]: If it's more than 1000ms then `true` for all servers will still mark `false` for each iteration. At least 10s should be set.

File: js/client/modules/@arangodb/clusterstats.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    print("marking FAIL since it took to long");
+    state.state = false;
+  }
+  fs.append(outfn, JSON.stringify(state) + "\n");
+
+  sleep(1);
```
Comment (KVS85) [inline js/client/modules/@arangodb/clusterstats.js]: Additional overhead from another client will take place with such low timing. At least 10s should be used.

File: js/client/modules/@arangodb/process-utils.js [hunk 7]
```diff
@@ -1479,6 +1483,17 @@ function checkClusterAlive(options, instanceInfo, addArgs) {
...
     }
   }
+
+  if (options.cluster) {
+    print("spawning cluster health inspector");
+    internal.env.INSTANCEINFO = JSON.stringify(instanceInfo);
+    internal.env.OPTIONS = JSON.stringify(options);
+    let args = makeArgsArangosh(options);
+    args['javascript.execute'] = fs.join('js', 'client', 'modules', '@arangodb', 'clusterstats.js');
+    const argv = toArgv(args).concat(args);
```
Comment (KVS85) [inline js/client/modules/@arangodb/process-utils.js]: What's the reason to concat the same value after `toArgv`?

File: js/client/modules/@arangodb/process-utils.js [hunk 8]
```diff
@@ -1479,6 +1483,17 @@ function checkClusterAlive(options, instanceInfo, addArgs) {
       throw new Error('cluster startup timed out after 10 minutes!');
     }
   }
+
+  if (options.cluster) {
```
Comment (KVS85) [inline js/client/modules/@arangodb/process-utils.js]: Any suite that has intentional arangod suspending will face false status for such inspection.

**Pull request closed without merging**